reactive/kubernetes_master.py

```diff
@@ -168,6 +168,35 @@ def channel_changed():
     set_upgrade_needed()
 
 
+# Returns True if a is subnet of b
+# This method is copied from cpython as it is available only from
+# python 3.7
+# https://github.com/python/cpython/blob/3.7/Lib/ipaddress.py#L1000
+def is_subnet_of(a, b):
+    try:
+        # Always false if one is v4 and the other is v6.
+        if a._version != b._version:
+            raise TypeError("{} and {} are not of the same version".format(
+                a, b))
+        return (b.network_address <= a.network_address and
+                b.broadcast_address >= a.broadcast_address)
+    except AttributeError:
+        raise TypeError("Unable to test subnet containment "
+                        "between {} and {}".format(a, b))
+
+
+def is_service_cidr_expansion():
+    service_cidr_from_db = db.get('kubernetes-master.service-cidr')
+    if service_cidr_from_db:
+        current_service_cidr = ipaddress.ip_network(service_cidr_from_db)
+        new_service_cidr = ipaddress.ip_network(hookenv.config('service-cidr'))
+        if not is_subnet_of(current_service_cidr, new_service_cidr):
+            hookenv.log("WARN: New k8s service cidr not superset of old one")
+            return False
+
+    return True
```

Comment:
```suggestion
def is_service_cidr_expansion():
    service_cidr_from_db = db.get('kubernetes-master.service-cidr')
    if not service_cidr_from_db:
        return False
    current_service_cidr = ipaddress.ip_network(service_cidr_from_db)
    new_service_cidr = ipaddress.ip_network(hookenv.config('service-cidr'))
    if not is_subnet_of(current_service_cidr, new_service_cidr):
        hookenv.log("WARN: New k8s service cidr not superset of old one")
        return False

    return True
```

Per @Cynerva's comment, I think the non-frozen-CIDR case result should be inverted and handled explicitly in `freeze_service_cidr()`